src/main/java/org/broadinstitute/hellbender/utils/haplotype/Haplotype.java

```diff
@@ -197,9 +197,17 @@ public void setCigar( final Cigar cigar ) {
 
     public Haplotype insertAllele( final Allele refAllele, final Allele altAllele, final int refInsertLocation, final int genomicInsertLocation ) {
         // refInsertLocation is in ref haplotype offset coordinates NOT genomic coordinates
-        final int haplotypeInsertLocation = ReadUtils.getReadCoordinateForReferenceCoordinate(alignmentStartHapwrtRef, cigar, refInsertLocation, ClippingTail.RIGHT_TAIL, true);
-        final byte[] myBases = this.getBases();
-        if( haplotypeInsertLocation == -1 || haplotypeInsertLocation + refAllele.length() >= myBases.length ) { // desired change falls inside deletion so don't bother creating a new haplotype
+        final Pair<Integer, CigarOperator> haplotypeInsertLocationAndOperator = ReadUtils.getReadCoordinateForReferenceCoordinate(alignmentStartHapwrtRef, cigar, refInsertLocation);
+
+        // can't insert outside the haplotype or into a deletion
+        if( haplotypeInsertLocationAndOperator.getLeft() == ReadUtils.CLIPPING_GOAL_NOT_REACHED || !haplotypeInsertLocationAndOperator.getRight().consumesReadBases() ) {
+            return null;
+        }
+        final int haplotypeInsertLocation = haplotypeInsertLocationAndOperator.getLeft();
+        final byte[] myBases = getBases();
+
+        // can't insert if we don't have any sequence after the inserted alt allele to span the new variant
+        if (haplotypeInsertLocation + refAllele.length() >= myBases.length) {
```

Comment:
Is this checking that there are non-variant bases (i.e. bases that are not in the variant context) on either side of the inserted allele?

Reply:
If the ref allele has n bases, you can think of the alt allele as deleting n - 1 bases (eg ACT -> A) followed, optionally by replacing those n - 1 bases (in the case of a MNP eg ACT -> A -> AGG).  Either way, if the haplotype doesn't contain those n - 1 bases it doesn't make sense to perform this transformation.

That is, suppose our haplotype is . . .ATTG.  Then we can insert an ATT -> A deletion by removing the TT, but we can't insert an ATTGC -> A deletion because the C is not there.